spec/controllers/users_controller_spec.rb

```diff
@@ -28,4 +32,51 @@
       get :download_pdf, id: 1                            
     end
   end
-end
\ No newline at end of file
+
+  describe "GET /users/:id/edit" do
+    it "should render the edit view for an admin user" do
+      sign_in admin
+      get :edit, id: tenant
+
+      expect(response.status).to eq(200)
+      expect(response).to render_template("edit")
+    end
+
+    it "should redirect a non-admin user to the root path" do
+      sign_in tenant
+      get :edit, id: tenant
+
+      expect(response).to redirect_to(root_path)
+
+      sign_in lawyer
+      get :edit, id: tenant
+
+      expect(response).to redirect_to(root_path)
+    end
+  end
+
+  describe "GET /users/:id/update" do
+    let(:params) { { id: tenant, user: { first_name: "Updated" } } }
+
+    it "should update the resource if the request comes from an admin user" do
+      admin.collaborations.create(collaborator: tenant)
```

Comment:
Oh in that case, should the edit action also be checking that you are in the User's collaborations association even for admins? Was the intent to allow the lawyer permission set to be able to access the edit page to if it had a collaboration. I had assumed that it was admin only because the form allowed you to set the User's permission level to admin, lawyer, or tenant.